### PR TITLE
Support using local hashicorp consul/vault images

### PR DESCRIPTION
**Pull request by priteau**
This requires stackhpc.hashicorp 2.4.0.

### CODE REVIEW

**Review by mnasiadka — DISMISSED** [review]

File: etc/kayobe/ansible/vault-deploy-overcloud.yml [hunk 1]
```diff
@@ -60,10 +60,15 @@
     - import_role:
         name: stackhpc.hashicorp.vault
       vars:
+        hashicorp_registry_url: "{{ overcloud_hashicorp_registry_url }}"
```
Comment (markgoddard) [inline etc/kayobe/ansible/vault-deploy-overcloud.yml:63]: Would it make sense to do the same for the seed vault to keep them in sync?

Comment (priteau) [reply etc/kayobe/ansible/vault-deploy-overcloud.yml:63]: I have updated the pull request to support local images for the seed. Can you please review again?

**Review by jovial — APPROVED** [review]
Looks reasonable to me. Only thought was that if we mirror the images from dockerhub directly, instead of pulling via ark, that we might end of using different versions of the images at different deployments. Not sure if they rebuild the images with the same tags?

**Comment by priteau** [comment]
> Looks reasonable to me. Only thought was that if we mirror the images from dockerhub directly, instead of pulling via ark, that we might end of using different versions of the images at different deployments. Not sure if they rebuild the images with the same tags?

I followed the approach we use for Ceph images. From what I can see on Docker Hub, if we switched to patch-level version (e.g. 1.14.1 instead of 1.14) we should get the same images:

* 1.14.0 was pushed on June 21 (day of 1.14.0 release)
* 1.14.1 was pushed on July 25 (when it was released)
* 1.14 was updated to point to 1.14.1

**Comment by mnasiadka** [comment]
I think it still would be better to sync those to Ark for internet-constrained environments (and counting docker hub pull limits)

**Review by markgoddard — APPROVED** [review]

**Comment by markgoddard** [comment]
> I think it still would be better to sync those to Ark for internet-constrained environments (and counting docker hub pull limits)

Perhaps, but it could be done as a separate task.

**Pull request merged**